presto-main-base/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -957,6 +957,9 @@ public PlanOptimizers(
                         costCalculator,
                         ImmutableSet.of(new RemoveRedundantIdentityProjections(), new PruneRedundantProjectionAssignments())));
 
+        // Pass after connector optimizer, as it relies on connector optimizer to identify empty input tables and convert them to empty ValuesNode
+        builder.add(new SimplifyPlanWithEmptyInput());
```

Comment:
does it need PruneUnreferencedOutputsToo like after the logical connector PlanOptimizers?

Reply:
The PruneUnreferencedOutputs works only for outer joins which are converted to projection when the inner side is empty, which in some cases the join keys are not in output and need to be pruned. It does not affect validity of the plan, but will further simplify the plan to allow later optimizer simplify the plan further. Since this instance of SimplifyPlanWithEmptyInput is running at the very end, and the PruneUnreferencedOutputs optimizer does not handle some nodes for example merge join node which may be in the plan by this stage, not adding it should be fine here (and can draft another PR to fix the PruneUnreferencedOutputs if needed).